network/network.go

```diff
@@ -611,70 +611,87 @@ func (n *Network) PeerDiagnostics() map[transport.PeerID]transport.Diagnostics {
+			return nil, fmt.Errorf("reprocess abort on transaction lookup, clock range [%d, %d): %w", offset, end, err)
 		}
 
-		lastLC := uint32(999)
-		for i := uint32(0); (lastLC+uint32(1))%batchSize == 0; i++ {
-			start := i * batchSize
-			end := start + batchSize
-			txs, err := n.state.FindBetweenLC(ctx, start, end)
-			if err != nil {
-				log.Logger().
-					WithError(err).
-					Errorf("Failed to reprocess transactions (start: %d, end: %d)", start, end)
-				return
+		for _, tx := range txs {
+			if tx.PayloadType() != contentType {
+				continue // filter
 			}
 
-			for _, tx := range txs {
-				if tx.PayloadType() == contentType {
-					// add to Nats
-					subject := fmt.Sprintf("%s.%s", events.ReprocessStream, contentType)
-					payload, err := n.state.ReadPayload(ctx, tx.PayloadHash())
-					if err != nil {
-						log.Logger().
-							WithError(err).
-							WithField(core.LogFieldTransactionRef, tx.Ref()).
-							WithField(core.LogFieldEventSubject, subject).
-							Error("Failed to publish transaction")
-						return
-					}
-					twp := events.TransactionWithPayload{
-						Transaction: tx,
-						Payload:     payload,
-					}
-					data, _ := json.Marshal(twp)
-					log.Logger().
-						WithField(core.LogFieldTransactionRef, tx.Ref()).
-						WithField(core.LogFieldEventSubject, subject).
-						Trace("Publishing transaction")
-					_, err = js.PublishAsync(subject, data)
-					if err != nil {
-						log.Logger().
-							WithError(err).
-							WithField(core.LogFieldTransactionRef, tx.Ref()).
-							WithField(core.LogFieldEventSubject, subject).
-							Error("Failed to publish transaction")
-						return
-					}
-				}
-				lastLC = tx.Clock()
+			// add to Nats
+			subject := fmt.Sprintf("%s.%s", events.ReprocessStream, contentType)
+			payload, err := n.state.ReadPayload(ctx, tx.PayloadHash())
+			if err != nil {
+				return nil, fmt.Errorf("reprocess abort on transaction %#x payload %#x: %w", tx.Ref(), tx.PayloadHash(), err)
+			}
+			twp := events.TransactionWithPayload{
+				Transaction: tx,
+				Payload:     payload,
+			}
+			data, _ := json.Marshal(twp)
+			log.Logger().
+				WithField(core.LogFieldTransactionRef, tx.Ref()).
+				WithField(core.LogFieldEventSubject, subject).
+				Trace("Publishing transaction")
+			_, err = js.PublishAsync(subject, data)
+			if err != nil {
+				return nil, fmt.Errorf("reprocess abort on transaction %#x publish: %w", tx.Ref(), err)
 			}
+		}
 
-			// give some time for Update transactions that require all read transactions to be closed
-			time.Sleep(time.Second)
+		if len(txs) == 0 || int(uint(txs[len(txs)-1].Clock())) < end-1 {
+			break
 		}
-	}()
+
+		// Workaround Nuts stoabs package which locks updates on any pending read
+		// transactions.
+		time.Sleep(time.Second)
+	}
+
+	// flush publish queue
+	select {
+	case <-js.PublishAsyncComplete():
+		break
+	case <-ctx.Done():
+		return nil, fmt.Errorf("reprocess terminate during final flush: %w", ctx.Err())
```

Comment:
This error will be logged and viewed by a system operator. What does this mean for him? Failed, should he/she do it again?

Reply:
Exactly the point of the message. 😅 It is impossible to tell. Maybe it worked, or maybe it didn't.

Reply:
The `final flush` is an internal concept which does not help a system operator. I would suggest a different wording.

Reply:
Yeah, bit weird now thinking about it. 😆 Perhaps "reprocess terminate while awaiting last enqueded"?